### ISSUE TITLE
Error in StreamProvider when loading two streams

### ISSUE THREAD

**Issue by uhensler** [polimediaupv/paella-core#132]
I am trying to load two videos with the following snippet.

```
{
  "metadata": {
    ...
  },
  "streams": [
    {
      "sources": {
        "hls": [
          {
            "src": "https://streaming.uibk.ac.at/public/_definst_/mp4:c102/c102267/Streaming-UIBK.mp4/playlist.m3u8",
            "mimetype": "video/mp4"
          }
        ]
      },
      "content": "stream1",
      "role": "mainAudio"
    },
    {
      "sources": {
        "mp4": [
          {
            "src": "http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4",
            "mimetype": "video/mp4"
          }
        ]
      },
      "content": "stream2",
      "role": null
    }
  ]
}
```

Unfortunately it does not work. The player does not show the videos. But I get an error message in the console.

```
Uncaught (in promise) TypeError: o is undefined
    t StreamProvider.js:366
    c StreamProvider.js:2
    _invoke StreamProvider.js:2
    m StreamProvider.js:2
    ct StreamProvider.js:2
    a StreamProvider.js:2
    promise callback*ct StreamProvider.js:2
    a StreamProvider.js:2
    promise callback*ct StreamProvider.js:2
    a StreamProvider.js:2
    lt StreamProvider.js:2
    lt StreamProvider.js:2
    value StreamProvider.js:357
    t StreamProvider.js:377
    c StreamProvider.js:2
    _invoke StreamProvider.js:2
    m StreamProvider.js:2
    ct StreamProvider.js:2
    a StreamProvider.js:2
    lt StreamProvider.js:2
    lt StreamProvider.js:2
    value StreamProvider.js:377
    t es.upv.paella.qualitySelector.js:27
    l runtime.js:65
    _invoke runtime.js:298
    x runtime.js:153
    s es.upv.paella.playbackRateButton.js:6
    a es.upv.paella.playbackRateButton.js:6
    promise callback*s es.upv.paella.playbackRateButton.js:6
    a es.upv.paella.playbackRateButton.js:6
    exports es.upv.paella.playbackRateButton.js:6
    exports es.upv.paella.playbackRateButton.js:6
    value es.upv.paella.qualitySelector.js:22
    t VideoContainer.js:144
    c VideoContainer.js:2
    _invoke VideoContainer.js:2
    m VideoContainer.js:2
    St VideoContainer.js:2
    a VideoContainer.js:2
    Ot VideoContainer.js:20
    Ot VideoContainer.js:2
    t VideoContainer.js:107
    t Plugin.js:63
    f Plugin.js:2
    _invoke Plugin.js:2
    w Plugin.js:2
    l Plugin.js:2
    a Plugin.js:2
    promise callback*l Plugin.js:2
    a Plugin.js:2
    promise callback*l Plugin.js:2
    a Plugin.js:2
    promise callback*l Plugin.js:2
    a Plugin.js:2
    promise callback*l Plugin.js:2
    a Plugin.js:2
    promise callback*l Plugin.js:2
    a Plugin.js:2

```

Strangely, the videos are loaded and displayed correctly when I swap the order of the two streams in the array.

**Comment by ferserc1** [comment]
The problem was caused by a bug when obtaining the list of available qualities for each video stream, when the second stream is not compatible with dynamic quality change, for that reason when you place the streams at the same time it works.

It has been fixed and will be released shortly, in version 1.8.5.

**Comment by uhensler** [comment]
Thank you!
